lib/cpp/src/thrift/server/TNonblockingServer.cpp

```diff
@@ -1520,13 +1524,9 @@ void TNonblockingIOThread::breakLoop(bool error) {
     ::abort();
   }
 
-  // sets a flag so that the loop exits on the next event
-  event_base_loopbreak(eventBase_);
+  // cause the loop to stop ASAP - even if it has things to do in it
+  event_base_loopexit(eventBase_, 0);
```

Comment:
+1
According to another look at the header doc, loopbreak is typically to be called from inside the loop.